src/app/components/Input/index.tsx

```diff
@@ -1,79 +1,105 @@
-  Pick<
-    HTMLInputProps,
-    'value' | 'type' | 'placeholder' | 'min' | 'max' | 'step' | 'onBlur'
-  >
-> & {
-  disabled?: boolean;
-  readOnly?: boolean;
-  className?: string;
+export type InputProps = Omit<InputBaseProps, 'ref'> & {
   classNameInput?: string;
-  dataActionId?: string;
-  onChange: (value: string) => void;
+  invalid?: boolean;
+  unit?: React.ReactNode;
 };
 
-export const Input: React.FC<InputProps> = ({
-  className,
-  classNameInput,
-  type,
-  step,
-  dataActionId,
-  onChange,
-  ...rest
-}) => {
-  const inputRef = useRef<HTMLInputElement>(null);
+export const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ className, classNameInput, type, invalid, unit, ...rest }, ref) => {
+    const inputRef = useRef<HTMLInputElement>(null) as React.MutableRefObject<
+      HTMLInputElement
+    >;
 
-  const onChangeWrapper = useCallback(event => onChange(event.target.value), [
-    onChange,
-  ]);
+    useImperativeHandle(ref, () => inputRef.current);
 
-  const onStepUp = useCallback(
-    event => {
-      inputRef.current?.stepUp();
-      onChange(inputRef.current?.value || '');
-    },
-    [onChange],
-  );
-  const onStepDown = useCallback(
-    event => {
-      inputRef.current?.stepDown();
-      onChange(inputRef.current?.value || '');
-    },
-    [onChange],
-  );
+    const onStepUp = useCallback(
+      event => {
+        inputRef.current?.stepUp();
+        rest.onChangeText?.(inputRef.current?.value || '');
+        rest.onChange?.(event);
+      },
+      [rest],
+    );
 
-  return (
-    <div className={classNames('tw-relative', className)}>
-      <input
-        ref={inputRef}
-        className={classNames(styles.input, classNameInput)}
-        type={type}
-        step={step}
-        data-action-id={dataActionId}
-        onChange={onChangeWrapper}
-        {...rest}
-      />
-      {type === 'number' && step ? (
-        <>
-          <button
-            className={classNames(styles.stepButton, styles.up)}
-            onClick={onStepUp}
-          >
-            <span />
-          </button>
-          <button
-            className={classNames(styles.stepButton, styles.down)}
-            onClick={onStepDown}
+    const onStepDown = useCallback(
+      event => {
+        inputRef.current?.stepDown();
+        rest.onChangeText?.(inputRef.current?.value || '');
+        rest.onChange?.(event);
+      },
+      [rest],
+    );
+
+    const wrapperClasses = useMemo(() => {
+      return {
+        'tw-border-sov-white': rest.readOnly,
+        'tw-border-warning tw-bg-warning':
+          typeof invalid !== 'undefined' && invalid,
+        'tw-border-success tw-bg-success':
+          typeof invalid !== 'undefined' && !invalid,
+      };
+    }, [invalid, rest.readOnly]);
+
+    return (
+      <div
+        className={classNames(
+          'tw-w-full tw-rounded-lg tw-max-w-xs tw-border tw-flex tw-flex-row tw-justify-between tw-items-center',
+          wrapperClasses,
```

Comment:
Can you move the classes to the index.module.scss please, otherwise anything passed in with the classname prop will be in conflict with the tailwind classes.